src/System.Buffers/src/System/Buffers/DefaultArrayPoolBucket.cs

```diff
@@ -80,7 +98,17 @@ internal void Return(T[] buffer)
                 // we can just drop this buffer
                 if (_index != 0)
                 {
+                    // If we have a debugger attached, do an extra safety check
+                    // to prevent a double-return scenario
+                    if (System.Diagnostics.Debugger.IsAttached)
```

Comment:
Ok, thanks.  "relatively cheap" may be something to still be wary of given that these Return/Rent methods are supposed to be as lightweight as possible.  I wonder whether this extra checking is really worthwhile.  There are many potential kinds of errors, e.g. using a buffer after it's been returned, returning a buffer multiple times, not returning a buffer, etc., and this only checks for a portion of those, and only in special cases.  It might instead be better to leave all such checking out of the library, and focus instead of providing appropriate post-processing of the ETW logs to be able to detect more such cases.  Evaluation of the buffer pool usage could potentially be built in to PerfView, both in terms of analysis of how buffers are being used (correctly) as well as flagging errors in usage.